docs/etsy.md

```diff
@@ -0,0 +1,430 @@
+| `urn:etsy:shop_id` | `shop_id` | User's shop ID |
+
+### Detailed User Information claims
+
+Endpoint: [`/v3/application/users/{user_id}` `getUser`](https://developers.etsy.com/documentation/reference#operation/getUser)
+
+#### Automapped claims
+
+_Requires `EtsyAuthenticationOptions.IncludeDetailedUserInfo = true`_
+
+| Claim Type | JSON Key | Auto-mapped |
+|:--|:--:|:--:|
+| `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress` | `primary_email` | ✓ |
+| `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname` | `first_name` | ✓ |
+| `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname` | `last_name` | ✓ |
+| `urn:etsy:image_url` | `image_url_75x75` | [Manual](#manually-added-claims) |
+
+> [!WARNING]
+> As those claims are set in Provider side `PostConfigureOptions`, you have to include them yourself if you bind from `PostConfigure<EtsyAuthenticationOptions>` also.
+
+#### Manually Added Claims
+
+The `image_url_75x75` claim is not auto-mapped to reduce data bloat. You can add it manually via either:
+
+**Direct JSON key mapping:**
+
+This sample does also work for regular JSON key mapping:
+
+```csharp
+options.ClaimActions.MapJsonKey(EtsyAuthenticationConstants.Claims.ImageUrl, "image_url_75x75");
+```
+
+**Claim Image using predefined extension method:**
+
+```csharp
+options.ClaimActions.MapImageClaim();
+```
+
+## Advanced Configuration
+
+The Etsy authentication handler can be configured in code or via configuration files.
+
+> [!NOTE]
+> Always make sure to use proper [Secret Management for production applications](https://learn.microsoft.com/aspnet/core/security/app-secrets).
+
+You can keep using code-based configuration, or bind from configuration values.
+
+> [!WARNING]
+> Avoid setting `UsePkce` from configuration, as Etsy requires PKCE for all OAuth flows.
+
+Here is a comprehensive `appsettings.json` example covering supported options and common scopes:
+
+```json
+{
+  "Etsy": {
+    "ClientId": "your-etsy-api-key",
+    "IncludeDetailedUserInfo": true,
+    "DetailedUserInfoEndpoint": "https://openapi.etsy.com/v3/application/users/{0}",
+    "AuthorizationEndpoint": "https://www.etsy.com/oauth/connect",
+    "TokenEndpoint": "https://openapi.etsy.com/v3/public/oauth/token",
+    "UserInformationEndpoint": "https://openapi.etsy.com/v3/application/users/me",
+    "CallbackPath": "/signin/etsy",
+    "SaveTokens": true,
+    "Scopes": [ "shops_r", "email_r" ]
+  }
+}
+```
+
+> [!NOTE]
+> We recommend saving tokens (`SaveTokens = true`) to facilitate token refresh, so the user does not need to re-authenticate frequently.
+> [!NOTE]
+> If `IncludeDetailedUserInfo` is set to `true` and the scopes `shops_r` and `email_r` scopes are sufficient, you don't need to set additional scopes in `appsettings.json`, they are added automatically.
+> [!TIP]
+> We recommend using the `EtsyAuthenticationDefaults` class in your `.AddEtsy` call which contains the default endpoint URLs.
```

Comment:
I don't think we need these recommendations - people can make their own decisions on how they way to integrate things.